lib/AST/Decl.cpp

```diff
@@ -6726,6 +6726,21 @@ void AbstractFunctionDecl::setBody(BraceStmt *S, BodyKind NewBodyKind) {
   }
 }
 
+void AbstractFunctionDecl::setBodyToBeReparsed(SourceRange bodyRange) {
+  assert(bodyRange.isValid());
+  assert(getBodyKind() == BodyKind::None ||
+         getBodyKind() == BodyKind::Skipped ||
```

Comment:
To make sure I understand this invariant: why is code-completion expecting `None` and `Skipped` rather than `Unparsed`?  Before we started reusing the ASTContext, I understand that "skipped" would make sense, but now that we may go back and parse it later, why is it not `Unparsed`?  And how can `None` happen? Does that means the user changed `func foo()` to `func foo() {}`?

Reply:
Good point.
`None` is not necessary here. `func foo()` -> `func foo() { #^A^# }` changes the interface cache and is not supported in fast-completion. Will remove it.
Also, non-target functions are marked as `Skipped` without a reason. I'm going to stop marking them `Skipped` and change this assertion to 

```cpp
  assert(getBodyKind() == BodyKind::Unparsed ||
         getBodyKind() == BodyKind::Parsed ||
         getBodyKind() == BodyKind::TypeChecked);
```

Reply:
Done